### PR TITLE
Support creating and using indices

### PR DESCRIPTION
**Pull request by xuebinsu**
Indexing is essential for fast data searching in database. 
This patch supports get access to database operators, 
creating and using indices in database.

### CODE REVIEW

File: tests/test_index.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from typing import List
+
+import pytest
+
+import greenplumpython as gp
+from tests import db
+
+
+def test_op_on_consts(db: gp.Database):
+    regex_match = gp.operator("~")
```
Comment (beeender) [inline tests/test_index.py:10]: can we predefine those operators? Those are standard PG operators I think?

Comment (beeender) [reply tests/test_index.py:10]: Oh, right, the "standard python operators" are handled in another way.

> Which ones do you think we should predefine

No idea, probably some common ones, like regex matching. But it is up to you want to add those in the coming release. I am fine with both.

File: requirements-doc.txt [hunk 2]
```diff
@@ -5,3 +5,4 @@ nbsphinx
 ipython~=8.10
 pydocstyle==6.2.3
 pandoc
+urllib3<2.0   # on Rocky 8: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'OpenSSL 1.0.2k-fips
```
Comment (yihong0618) [inline requirements-doc.txt:8]: nice

**Review by yihong0618 — APPROVED** [review]
LGTM

**Review by ruxuez — APPROVED** [review]
LGTM

File: greenplumpython/op.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+"""
+Indices are essential for fast data searching. With indices, we can retrieve
+rows we want by scanning the index, rather than the entire dataframe. As a
+result, when the dataframe is large, the amount of data to be scanned is
+typically mich smaller with index.
```
Comment (ruxuez) [inline greenplumpython/op.py:5]:
```suggestion
typically much smaller with index.
```

File: greenplumpython/op.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+In pandas, a :code:`DataFrame` or :code:`Series` can have only one index. This
+means that we can do index scan based on one column. While for other columns,
+we will have to go through all rows. This can be rather inefficient.
+
+Backed by database systems, GreenplumPython overcomes this limitation by
+allowing creating
```
Comment (ruxuez) [inline greenplumpython/op.py:12]:
```suggestion
allowing the creation of
```

File: greenplumpython/op.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+- multiple indices, each for one set of columns, and
```
Comment (ruxuez) [inline greenplumpython/op.py:14]: On VMware Greenplum Documentation, the plural formal of index is **indexes**, maybe we should standardise with it.

File: greenplumpython/op.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+- multiple types of indices for same set of columns, each for one order.
+
+In this way, we can search a GreenplumPython's dataframe with index scan, on
+more than one column set in more than one order.
+
+For example, for a dataframe of embeddings generated by AI, it may contain
```
Comment (ruxuez) [inline greenplumpython/op.py:20]:
```suggestion
For example, a dataframe containing AI-generated embeddings may contain
```

File: greenplumpython/op.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+- a column of IDs and
+- a column of vectors.
+
+We can search for embeddings by either IDs and vectors with index scan after
```
Comment (ruxuez) [inline greenplumpython/op.py:25]:
```suggestion
We can either search for embeddings by IDs and vectors using index scan after
```

File: greenplumpython/op.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+distance. We can create two indices on the vector column, each for one
+similarity metric.
+
+How to search a dataframe with index is defined by a set of operators on the
+indexed columns. For example, when scanning a B-tree index, relational
+operators, such as :code:`>`, :code:`<`, and :code:`=`, are required for
+comparing two values. These operators are encapsulated as operator classes.
+Different data types have different operator classes for an index. For example,
+integers and floats are compared in different ways. Even for the same data type,
+We can change how two values compare by changing the operator class.
```
Comment (ruxuez) [inline greenplumpython/op.py:39]:
```suggestion
we can change how two values are compared by changing the operator class.
```

File: greenplumpython/op.py [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+Since indices depends on operators to work, to use index scan, we need to specify
```
Comment (ruxuez) [inline greenplumpython/op.py:41]:
```suggestion
Since indices depend on operators to work, to use index scan, we need to specify
```

**Review by ruxuez — COMMENTED** [review]
Maybe just some modifications and suggestions on doc

File: greenplumpython/op.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
+"""
```
Comment (beeender) [inline greenplumpython/op.py:1]: doc style check is not enabled for this new file, see https://github.com/greenplum-db/GreenplumPython/blob/b219b93483176227cdb40966fe4f4464bbf1ac2f/tox.ini#L73

I think we can enable it for all files under `greenplumpython` now.

**Review by beeender — APPROVED** [review]
LGTM. Please enable the doc style check in this PR or a later PR.

**Pull request merged**